frontend/src/modules/scaffold/views/scaffoldAssetView.js

```diff
@@ -209,96 +182,100 @@ define(function(require) {
     */
 
     focus: function() {
-      if (this.hasFocus) return;
-      /**
-      * makes input the `document.activeElement`, triggering this editor's
-      * focus` event, and setting `this.hasFocus` to `true`.
-      * See this.events above for more detail
-      */
-      this.$('input').focus();
+      if (!this.hasFocus) {
+        this.$('input').focus();
+      }
     },
 
     blur: function() {
-      if(this.hasFocus) this.$('input').blur();
+      if (this.hasFocus) {
+        this.$('input').blur();
+      }
     },
 
     onAssetButtonClicked: function(event) {
       event.preventDefault();
+
       Origin.trigger('modal:open', AssetManagementModalView, {
         collection: new AssetCollection,
-        assetType: this.schema.fieldType,
+        assetType: this.schema.inputType,
         _shouldShowScrollbar: false,
         onUpdate: function(data) {
-          if (!data) {
-            return;
-          }
+          if (!data) return;
+
           if ('heroImage' === this.key) {
             this.setValue(data.assetId);
             this.saveModel({ heroImage: data.assetId });
             return;
           }
+
+          var model = Origin.scaffold.getCurrentModel();
+
           var courseAssetObject = {
-            contentTypeId: Origin.scaffold.getCurrentModel().get('_id') || '',
-            contentType: Origin.scaffold.getCurrentModel().get('_type'),
-            contentTypeParentId: Origin.scaffold.getCurrentModel().get('_parentId') || Origin.editor.data.course.get('_id'),
+            contentTypeId: model.get('_id') || '',
+            contentType: model.get('_type'),
+            contentTypeParentId: model.get('_parentId') || Origin.editor.data.course.get('_id'),
             fieldname: data.assetFilename,
             assetId: data.assetId
           };
+
           // all ScaffoldAssetViews listen to the autofill event, so we trigger
           // that rather than call code directly
           // FIXME only works with graphic components
```

Comment:
Any chance of looking at this FIXME while your here?

Reply:
The associated code it's referring to is actually part of [Asset Management](https://github.com/adaptlearning/adapt_authoring/blob/develop/frontend/src/modules/assetManagement/views/assetManagementModalView.js#L19). Would you like me to remove the comment?

Reply:
May as well if its not relevant anymore.